src/asyncOperationStateUtils.js

```diff
@@ -52,10 +55,44 @@ const updateAsyncOperationDescriptor = (state, descriptorOptions) => {
   };
 };
 
+const getLatestOperationByDescriptorId = (state, descriptorId) => {
+  return reduce(state.operations, (acc, operation) => {
+    if (operation.descriptorId === descriptorId) {
+      if (!acc) {
+        return operation;
+      }
+      if (acc.lastFetchStatusTime <= operation.lastFetchStatusTime) {
+        return operation;
+      }
+      return acc;
+    }
+  }, {});
+};
+
+const setInvalidatedOperationByKeyState = (state, asyncOperationKey, descriptorId) => {
+  const asyncOperationDescriptor = state.descriptors[descriptorId];
+
+  return {
+    ...state,
+    operations: {
+      ...state.operations,
+      [asyncOperationKey]: asyncOperationDescriptor.operationType === ASYNC_OPERATION_TYPES.READ
+        ? initialReadAsyncOperationForAction(asyncOperationDescriptor.descriptorId, asyncOperationKey)
+        : initialWriteAsyncOperationForAction(asyncOperationDescriptor.descriptorId, asyncOperationKey),
+    },
+  };
+};
+
 // This function will do all the work to determine if an async operation is returned as an initial async operation
 // (if it is not found in state), an asyncOperation with parentAsyncOperation metaData (recursively searched to find if the parentAsyncOperation is more
 // up-to-date) or just the asyncOperation itself if the none of the above apply.
-const getAsyncOperation = (state, asyncOperationKey, asyncOperationDescriptor, asyncOperationParams, fieldsToAdd) => {
+const getAsyncOperationFromState = ({
```

Comment:
What's the purpose of this change? It seems like it would be unnecessarily breaking.

Reply:
it changes the name so it doesn't conflict with `getAsyncOperation` that is exported by asyncOperationManagerUtils which is the actually api exposed to the consumer.